src/data_processing/custom_python_package.py

```diff
@@ -0,0 +1,610 @@
+        lat_lon_df = df[["latitude", "longitude", "pixel_geom_id"]].copy()
+        lat_lon_df = lat_lon_df.drop_duplicates()
+        admin_df = gpd.read_file(admin_boundary_file_path)
+
+        # Computes a reference grid by linking grid points to administrative
+        # boundaries. Only done once as all the ensemble models use the same
+        # spatial grid
+        if batch_number == 0:
+            grid_df = _create_reference_grid(
+                lat_lon_df, admin_df, admin_code_label
+            )
+
+        # Link_df is a MxN link table between grid points and administrative
+        # boundaries. The groupby allows to drop grid points duplicate in
+        # the first case or aggregate into admin boundaries in the second one
+        link_df = pd.merge(df, grid_df, on="pixel_geom_id")
+        batch_data_grid_df = (
+            link_df.groupby(
+                [
+                    "pixel_geom_id",
+                    "latitude",
+                    "longitude",
+                    "number",
+                    "valid_time_year",
+                    "valid_time_month",
+                    "lead_time",
+                ]
+            )["tp_mm_day"]
+            .mean()
+            .reset_index()
+        )
+        batch_data_adm_df = (
+            link_df.groupby(
+                [
+                    "adm1_pcode",
+                    "number",
+                    "valid_time_year",
+                    "valid_time_month",
+                    "lead_time",
+                ]
+            )["tp_mm_day"]
+            .mean()
+            .reset_index()
+        )
+
+        # Concatenate all ensemble model into a single DataFrame
+        if batch_number == 0:
+            data_grid_df = batch_data_grid_df
+            data_adm_df = batch_data_adm_df
+        else:
+            data_grid_df = pd.concat([data_grid_df, batch_data_grid_df])
+            data_adm_df = pd.concat([data_adm_df, batch_data_adm_df])
+
+    data_grid_df.reset_index(inplace=True, drop=True)
+    data_adm_df.reset_index(inplace=True, drop=True)
+
+    # Export data to parquet file
+    grid_df.to_parquet(ref_grid_file_path, compression="gzip")
+    data_grid_df.to_parquet(pixel_output_file_path, compression="gzip")
+    data_adm_df.to_parquet(adm_output_file_path, compression="gzip")
+
+    print("data exported")
+
+
+def pre_process_era5_data(
+    era5_file_path: str,
+    ecmwf_file_path: str,
+    ref_grid_file_path: str,
+    pixel_output_file_path: str,
+    adm_output_file_path: str,
+):
+    """
+        Loads the ERA5 climate data grib file and converts it to a DataFrame.
+        Also adapt columns names, precipitation units and link grid points
+        to administrative boundaries. Finally, computes the average climatology
+        (per location and month) and export the resulting DataFrame
+        to a parquet file.
+
+
+    Parameters
+    ----------
+        input_file_path: str, Path to the ERA5 climate data grib file to be
```

Comment:
@ddebarros-mapaction please update the doc-string and replace the redundant `input_file_path` with `era5_file_path` and `ecmwf_file_path`